build/Makefile

```diff
@@ -384,6 +384,7 @@ STATIC_PROVIDERLIB_SRCFILES += \
 	$(PROVIDER_DIR)/support/mycnf-paths.cpp \
 	$(PROVIDER_DIR)/support/productdependencies.cpp \
 	$(PROVIDER_DIR)/support/sqlauth.cpp \
+	$(PROVIDER_DIR)/support/mycnf-scanner.cpp \
```

Comment:
The problem is that the provider can't reliably to this. It is platform specific, but on some platforms, the MySQL configuration files are not W:R. Since the provider runs as-requestor, that means that a non-root user would not be able to see MySQL stats, and that wasn't acceptable to Kris Bash, our PM.

As a result, the MySQL configuration files are read exclusively by omi_preexec, which runs with privileges. That ends up setting up the auth file, allowing the provider to run without root privileges. If sqlbinding.cpp is reading the config files, that's an error, I believe, and one that you need to track down. Remember, this code used to live exclusively in omi_preexec, so sqlbinding.cpp reading the config files wasn't previously possible (I believe).

Since the provider itself should not read the cnf files (and indeed, on some platforms, CAN'T read the conf files), the provider itself should not be accessing these routines. It should be harmless to remove them.

What sqlbinding.cpp DOES do is read the auth file to get the sql ports. But that's different than reading the cnf files.

---
In reply to: [66894783](https://github.com/Microsoft/MySQL-Provider/pull/5#discussion_r66894783) [](ancestors = 66894783)